spec.emu

```diff
@@ -385,12 +382,24 @@ location: https://github.com/tc39/proposal-measure/
     <p>It performs the following steps when called:</p>
     <emu-alg>
       1. Let _O_ be the *this* value.
-      1. Perform ? RequireInternalSlot(_O_, [[AmountData]]).
-      1. If _options_ is not an Object, throw a *TypeError* exception.
-      1. Let _v_ be _O_.[[AmountData]].
-      1. Let _s_ be Number::toString(_v_, 10).
-      1. Return ? Call (%Amount%, _O_, « _s_, _options_ »).
-      1. Return OrdinaryObjectCreate(*"%Amount.prototype%"*, « [[AmountData]], [[SignificantDigits]], [[FractionDigits]], [[Unit]] »).
+      1. Perform ? RequireInternalSlot(_O_, [[Value]]).
+      1. Let _options_ be ? GetOptionsObject(_options_).
+      1. If ? HasOwnProperty(_options_, *"currency"*), throw a *TypeError* exception.
+      1. If ? HasOwnProperty(_options_, *"unit"*), throw a *TypeError* exception.
```

Comment:
These need to be conditional on _O_.[[Currency]] or _O_.[[Unit]] being a string. Adding a currency or unit to an Amount that doesn't have either ought to be fine.